src/Nethermind/Nethermind.Specs/ChainSpecStyle/ChainSpecBasedSpecProvider.cs

```diff
@@ -149,7 +149,7 @@ private static ForkActivation[] CreateTransitionActivations(SortedSet<long> tran
 
             foreach (ulong timestamp in transitionTimestamps)
             {
-                transitionActivations[index++] = new ForkActivation(biggestBlockTransition, timestamp);
+                transitionActivations[index++] = new ForkActivation(++biggestBlockTransition, timestamp);
```

Comment:
Why do we need it?

Reply:
Because our GetSpec method usies binary search. And sometimes when the blocknumber only fork and timestamp gork has the same blocknumber it takes the timestamp one onstead of the blocknumber one

Reply:
Can you provide a more detailed explanation with examples?

Reply:
if you remove the `++` and run chainspecbasedspecproviderTests goerli will fail. the search function will return the shanghai spec for london, just becuase london and shanghai have the same blockNumber (biggestBlockNumber)

Reply:
yeah and it fails on... london test :)
Will provide better test separation shortly

Reply:
Removed 2nd `++`, added a comment on first `++` .

Reply:
@smartprogrammer93 isn't a test issue assumption issue? we can assume in tests that London and Shangai will have different block numbers